### PR TITLE
Added support of integer type suffixes "u" and "l"

### LINKED ISSUE #62: Support all integer literals
Like "u" for unsigned, e..g `4u` (being equivalent to `(uint)4`). See: https://msdn.microsoft.com/en-us/library/aa664674(v=vs.71).aspx

### PR DESCRIPTION
**Pull request by metoule**
This fixes #62 by adding support of integer type suffixes "u" and "l" (and their combination UL, LU, etc).
It's now possible to use `123u`, `123L`, etc.

I also added support of real type suffix "d" so that `123d` is properly parsed as a double.

### CODE REVIEW

File: src/DynamicExpresso.Core/Parsing/Parser.cs [hunk 1]
```diff
@@ -639,18 +639,35 @@ private Expression ParseIntegerLiteral()
 		{
 			ValidateToken(TokenId.IntegerLiteral);
 			var text = _token.text;
+
+			var isUnsigned = false;
+			var isLong = false;
+			var numberEnd = text.Length - 1;
+			for (; numberEnd >= 0; numberEnd--)
```
Comment (davideicardi) [inline src/DynamicExpresso.Core/Parsing/Parser.cs:646]: Why the `for` here? Just using `text[numberEnd]` is not enought? Sorry probably I missing something obvious or don't remember ;-)

Comment (davideicardi) [reply src/DynamicExpresso.Core/Parsing/Parser.cs:646]: I ok get it! It is for supporting `UL` suffix or similar ... Right?

**Review by davideicardi — CHANGES_REQUESTED** [review]
Thank you for the great work!

Maybe we can also test what happen for invalid suffix with an unit test?
For example I would expect a parse exception for something like `4325FL` or `3dl`? Right? And also for unknown suffix, like `4G`?

Also can you please just add a little section in documentation with all the supported numeric literals, maybe extending this one https://github.com/davideicardi/DynamicExpresso#literals ?

File: src/DynamicExpresso.Core/Parsing/Parser.cs (hunk 1 above)
Comment (metoule) [reply src/DynamicExpresso.Core/Parsing/Parser.cs:646]: Yes, it's because of the UL suffixes: there can be several characters to the suffix :)

**Review by davideicardi — APPROVED** [review]
Great!

**Pull request merged**